### PR TITLE
allow nested defers at the same level

### PR DESCRIPTION
**Pull request by yaacovCR**
Implements Option A from https://github.com/graphql/defer-stream-wg/discussions/80

### CODE REVIEW

**Comment by yaacovCR** [comment]
@github-actions publish-pr-on-npm

**Comment by yaacovCR** [comment]
This is the starting point for #4026, a major rewrite of incremental delivery with the aim of following the spec edits at https://github.com/graphql/graphql-spec/pull/1077

Considering that, if we are confident #4026 will get merged, this PR becomes important less for the implementation, and more for the test changes.

File: src/execution/__tests__/defer-test.ts [hunk 1]
```diff
@@ -472,6 +476,40 @@ describe('Execute: defer directive', () => {
...
+        data: {
+          hero: {},
+        },
+        pending: [{ id: '0', path: ['hero'] }],
+        hasNext: true,
+      },
+      {
+        pending: [{ id: '1', path: ['hero'] }],
+        completed: [{ id: '0' }],
+        hasNext: true,
```
Comment (robrichard) [inline src/execution/__tests__/defer-test.ts:503]: this payload is unexpected to me. Since the outer @defer has no fields, shouldn't it be skipped?

Comment (robrichard) [reply src/execution/__tests__/defer-test.ts:503]: Do we need to merge https://github.com/graphql/graphql-js/pull/3993 first?

Comment (yaacovCR) [reply src/execution/__tests__/defer-test.ts]: Done => and rebased.

**Review by robrichard — APPROVED** [review]

**Review by 5twelve — APPROVED** [review]

**Pull request merged**